pkg/core/dao/dao.go

```diff
@@ -89,6 +89,11 @@ func (dao *Simple) GetWrapped() *Simple {
 	return d
 }
 
+// GetUnwrapped returns the underlying DAO. It does not perform changes persist.
+func (dao *Simple) GetUnwrapped() *Simple {
```

Comment:
The problem is that we have to store the old DAO state in case if revert is needed. Currently there's only single DAO inside the interop context, and it's very convenient. Othervise we have to add some `oldDAO` field inside the interop context and be careful with its usages.